src/fortran/client.F90

```diff
@@ -139,32 +139,45 @@ module smartredis_client
 contains
 
 !> Initializes a new instance of a SmartRedis client
-subroutine initialize_client( self, cluster )
+function initialize_client(self, cluster)
+  integer(kind=enum_kind)           :: initialize_client
   class(client_type), intent(inout) :: self    !< Receives the initialized client
   logical, optional,  intent(in   ) :: cluster !< If true, client uses a database cluster (Default: .false.)
 
+  ! Local variables
+  integer(kind=enum_kind) :: result
+
   if (present(cluster)) self%cluster = cluster
-  self%client_ptr = c_constructor(self%cluster)
-  self%is_initialized = .true. 
-end subroutine initialize_client
+  result = c_constructor(self%cluster, self%client_ptr)
```

Comment:
This applies to this function and the others: why not just store the result here and not at the end of the function? I realize that there are some things which look like very similar names i.e. the function name is `get_model` and it calls `get_model_c`, but that might be a case for changing the function names.
e.g.
```
initialize_client = c_constructor(self%cluster, self%client_ptr)
```

Reply:
Done

Reply:
Done